analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/ProjectBuilder.cs

```diff
@@ -65,15 +65,15 @@ public ProjectBuilder AddReferences(IEnumerable<MetadataReference> references)
         public ProjectBuilder AddProjectReference(Func<SolutionBuilder, ProjectId> getProjectId) =>
             FromProject(project.AddProjectReference(new ProjectReference(getProjectId(Solution))));
 
-        public ProjectBuilder AddDocuments(IEnumerable<string> paths) =>
-            paths.Aggregate(this, (projectBuilder, path) => projectBuilder.AddDocument(path));
+        public ProjectBuilder AddDocuments(IEnumerable<string> paths, string basePath = "") =>
+            paths.Aggregate(this, (projectBuilder, path) => projectBuilder.AddDocument(path, basePath));
 
-        public ProjectBuilder AddDocument(string path)
+        public ProjectBuilder AddDocument(string path, string basePath = "")
         {
             _ = path ?? throw new ArgumentNullException(nameof(path));
             var fileInfo = new FileInfo(path);
             return fileInfo.Extension == fileExtension
-                ? AddDocument(project, fileInfo.Name, File.ReadAllText(fileInfo.FullName, Encoding.UTF8))
+                ? AddDocument(project, Path.Combine(basePath ?? string.Empty, fileInfo.Name), File.ReadAllText(fileInfo.FullName, Encoding.UTF8))
```

Comment:
Actually, I don't like what the `basePath` means in this method.
We have path like `C:\Projects\sonar-dotnet\Analyzers\test\SonarAnalyzer.UniTest\TestCases\SubDir\File.cs`
What we really want, is to extract whatever is behind  `TestCases\` fragment to get `SubDir\File.cs`.
What it currently does is providing any artificial fake prefix, where caller can inject `Whatever\File.cs` that doesn't have anything in common with `SubDir\File.cs`.

Let's take `fileInfo.FullName`, search for `TestCases\` and use whatever follows. In case `IndeOf` returns -1, I think we can safely throw.

It's not very pure solution, but it's part of TestFramework that is bound to our test infra so we can afford such dependency. It will remove any possible confusion and offload the responsibility to a single place. So we don't have to deal with the entire invocation tree of this method.

`VerifierTest` will fail on this. That's fine, we can just add "TestCases" infix into `WriteFile` method to met the expectation.